app/Cells/MemberList.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\Cells;
+
+use App\Concerns\HasTableSearchComponent;
+use App\Models\UserModel;
+use CodeIgniter\Pager\Pager;
+use CodeIgniter\View\Cells\Cell;
+
+class MemberList extends Cell
+{
+    use HasTableSearchComponent;
+
+    protected string $view = 'member_list_cell';
+
+    protected array $members;
+    protected Pager $pager;
+
+    public function mount()
+    {
+        $this->defineTableSearchComponent();
+
+        helper('form');
+
+        $model = model(UserModel::class);
+
+        $this->members = $model->searchMembers(
+            $this->search, $this->page, $this->perPage, $this->sortColumn, $this->sortDirection
+        );
+
+        $this->pager = $model->pager->setPath($this->baseURL);
+    }
+
+    protected function getMembersProperty(): array
+    {
+        return $this->members;
+    }
+
+    protected function validRoles(): array
+    {
+        return array_combine($this->validSearch['role'], $this->validSearch['role']);
```

Comment:
@michalsn 
`$this->validSearch['role']` produces the following string. This is not good, we need the **role title**.
```php
all => string (3) "all"
⇄superadmin => string (10) "superadmin"
⇄admin => string (5) "admin"
⇄developer => string (9) "developer"
⇄user => string (4) "user"
⇄beta => string (4) "beta"
```

**Hard Code**
```php
        return array_combine($this->validSearch['role'], [
            'all' =>  "all",
            'superadmin' => setting('AuthGroups.groups')['superadmin']['title'],
            'admin'      =>  setting('AuthGroups.groups')['admin']['title'],
            'developer'  => setting('AuthGroups.groups')['developer']['title'],
            'user'       =>  setting('AuthGroups.groups')['user']['title'],
            'beta'       =>  setting('AuthGroups.groups')['beta']['title'],
        ]);
```
Befor
![before](https://github.com/lonnieezell/forum-example/assets/9530214/ad883c80-03b6-4448-ac6f-5b2352a3f179)

![after](https://github.com/lonnieezell/forum-example/assets/9530214/b337bd9e-8314-40f4-8577-16f25bcd016e)

Reply:
It was on my to-do list for later, but since you mentioned it... I added it now. Thanks.

Reply:
Thank you!